plugins/woocommerce/src/Blocks/BlockTypes/BlockifiedProductDetails.php

```diff
@@ -35,69 +35,236 @@ protected function get_block_type_style() {
+		)[0];
+
+		foreach ( $template['innerBlocks'] as &$block ) {
+			if ( 'woocommerce/accordion-header' === $block['blockName'] ) {
+				$block['innerContent'] = array_map(
+					function ( $inner_content ) use ( $title ) {
+						return str_replace( '{{title}}', $title, $inner_content );
+					},
+					$block['innerContent']
+				);
+			}
+			if ( 'woocommerce/accordion-panel' === $block['blockName'] ) {
+				$block['innerBlocks']  = parse_blocks( $content );
+				$openning_tag          = reset( $block['innerContent'] );
+				$closing_tag           = end( $block['innerContent'] );
+				$block['innerContent'] = array_merge(
+					array( $openning_tag ),
+					array_fill( 0, count( $block['innerBlocks'] ), null ),
+					array( $closing_tag )
+				);
+			}
+		}
+
+		return $template;
+	}
+
+	/**
+	 * Inject parsed accordion blocks.
+	 *
+	 * @param array $parsed_block Parsed block.
+	 * @param array $accordion_blocks Accordion blocks.
+	 *
+	 * @return array Parsed block.
+	 */
+	private function inject_parsed_accordion_blocks( $parsed_block, $accordion_blocks ) {
+		if ( 'woocommerce/accordion-group' === $parsed_block['blockName'] ) {
+			$parsed_block['innerBlocks']  = array_merge( $parsed_block['innerBlocks'], $accordion_blocks );
+			$parsed_block['innerBlocks']  = array_values( array_filter( $parsed_block['innerBlocks'] ) );
+			$openning_tag                 = reset( $parsed_block['innerContent'] );
+			$closing_tag                  = end( $parsed_block['innerContent'] );
+			$parsed_block['innerContent'] = array_merge(
+				array( $openning_tag ),
+				array_fill( 0, count( $parsed_block['innerBlocks'] ), null ),
+				array( $closing_tag )
+			);
+			return $parsed_block;
+		}
+
+		foreach ( $parsed_block['innerBlocks'] as $key => $inner_block ) {
+			$parsed_block['innerBlocks'][ $key ] = $this->inject_parsed_accordion_blocks( $inner_block, $accordion_blocks );
+		}
+
+		return $parsed_block;
+	}
+
+	/**
+	 * Hide empty accordion items.
+	 *
+	 * @param array $parsed_block Parsed block.
+	 * @param array $context Context.
+	 *
+	 * @return array Parsed block.
+	 */
+	private function hide_empty_accordion_items( $parsed_block, $context ) {
+		if ( ! $this->has_accordion( $parsed_block ) ) {
+			return $parsed_block;
+		}
 
-		$counter = 0;
-		while ( $p->next_tag( array( 'class_name' => 'wp-block-woocommerce-accordion-item' ) ) ) {
-			if ( ! $accordion_items_visibility[ $counter ] ) {
-				$p->set_attribute( 'style', 'display:none;' );
-				$p->set_attribute( 'hidden', true );
+		if ( 'woocommerce/accordion-group' === $parsed_block['blockName'] ) {
+			foreach ( $parsed_block['innerBlocks'] as $key => $inner_block ) {
+				$parsed_block['innerBlocks'][ $key ] = $this->mark_accordion_item_hidden( $inner_block, $context );
 			}
-			++$counter;
+			$parsed_block['innerBlocks']  = array_values( array_filter( $parsed_block['innerBlocks'] ) );
+			$openning_tag                 = reset( $parsed_block['innerContent'] );
```

Comment:
I guess that there is a typo. Should it be: 
```suggestion
			$opening_tag                 = reset( $parsed_block['innerContent'] );
```

Reply:
Thanks! I fixed in [`fb66ee7` (#57973)](https://github.com/woocommerce/woocommerce/pull/57973/commits/fb66ee7e8daa0e73941b9beab3733dfd3acd5538).